sync.sh

```diff
@@ -0,0 +1,541 @@
+        for array in "${INPUT_ARRAY[@]}"; do
+            { [[ -d ${array} ]] && FINAL_INPUT_ARRAY+=("${array[@]}"); } || {
+                printf "\nError: Invalid Input ( %s ), no such directory.\n" "${array}"
+                exit 1
+            }
+        done
+    fi
+
+    mapfile -t FINAL_INPUT_ARRAY <<< "$(_remove_array_duplicates "${FINAL_INPUT_ARRAY[@]}")"
+}
+
+###################################################
+# Grab config variables and modify defaults if necessary
+# Globals: 5 variables, 2 functions
+#   Variables - INFO_PATH, UPDATE_DEFAULT_CONFIG, DEFAULT_ARGS
+#               UPDATE_DEFAULT_ARGS, UPDATE_DEFAULT_TIME_TO_SLEEP, TIME_TO_SLEEP
+#   Functions - _print_center, _update_config
+# Arguments: None
+# Result: source .info file, grab COMMAND_NAME and CONFIG
+#   source CONFIG, update default values if required
+###################################################
+_config_variables() {
+    if [[ -f "${INFO_PATH}/google-drive-upload.info" ]]; then
+        # shellcheck source=/dev/null
+        source "${INFO_PATH}/google-drive-upload.info"
+    else
+        _print_center "justify" "google-drive-upload is not installed system wide." "=" 1>&2
+        exit 1
+    fi
+
+    # Check if command exist, not necessary but just in case.
+    if ! type "${COMMAND_NAME}" &> /dev/null; then
+        printf "Error: %s is not installed, use -c/--command to specify.\n" "${COMMAND_NAME}" 1>&2
+        exit 1
+    fi
+
+    ARGS+=" -q "
+    SYNC_TIME_TO_SLEEP="3"
+    # Config file is created automatically after first run
+    if [[ -r ${CONFIG} ]]; then
+        # shellcheck source=/dev/null
+        source "${CONFIG}"
+        if [[ -n ${UPDATE_DEFAULT_CONFIG} ]]; then
+            printf "%s\n" "${CONFIG}" >| "${INFO_PATH}/google-drive-upload.configpath"
+        fi
+    fi
+    SYNC_TIME_TO_SLEEP="${TO_SLEEP:-${SYNC_TIME_TO_SLEEP}}"
+    ARGS+=" ${SYNC_DEFAULT_ARGS:-} "
+    "${UPDATE_DEFAULT_ARGS:-:}" DEFAULT_ARGS " ${ARGS} " "${CONFIG}"
+    "${UPDATE_DEFAULT_TIME_TO_SLEEP:-:}" SYNC_TIME_TO_SLEEP "${SYNC_TIME_TO_SLEEP}" "${CONFIG}"
+}
+
+###################################################
+# Triggers in case either -j & -k or -l flag ( both -k|-j if with positive integer as argument )
+# Priority: -j > -i > -l > -k
+# Globals: 5 variables, 6 functions
+#   Variables - JOB, SHOW_JOBS_VERBOSE, INFO_PID, LOG_PID, KILL_PID ( all array )
+#   Functions - _check_pid, _setup_loop_variables
+#               _kill_job, _show_jobs, _get_job_info, _remove_job
+# Arguments: None
+# Result: show either job info, individual info or kill job(s) according to set global variables.
+#   Script exits after -j and -k if kill all is triggered )
+###################################################
+_do_job() {
+    case "${JOB[*]}" in
+        *SHOW_JOBS*)
+            _show_jobs "${SHOW_JOBS_VERBOSE:-}"
+            exit
+            ;;
+        *PIDS*)
+            for pid in "${ALL_PIDS[@]}"; do
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This parent loop has its index variable overridden.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5619177)